src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/AbstractClassCouplingCheck.java

```diff
@@ -294,7 +409,24 @@ public void checkCoupling() {
          */
         private boolean isSignificant(String candidateClassName) {
             return !excludedClasses.contains(candidateClassName)
-                    && !candidateClassName.startsWith("java.lang.");
+                    && !isFromExcludedPackage(candidateClassName);
+        }
+
+        /**
+         * Checks if given class should be ignored as it belongs to excluded package.
+         * @param candidateClassName class to check
+         * @return true if we should not count this class.
+         */
+        private boolean isFromExcludedPackage(String candidateClassName) {
+            String classNameWithPackage = candidateClassName;
+            if (!candidateClassName.contains(DOT)) {
+                classNameWithPackage = parentContext
+                    .getClassNameWithPackage(candidateClassName)
+                    .orElse(null);
+            }
+
+            return classNameWithPackage != null && excludedPackages.stream()
+                .anyMatch(classNameWithPackage::startsWith);
```

Comment:
So if I say to exclude package `a.b` and we have an import from `a.b.c` package, it will be excluded since you use `startsWith` right?
IMO, we should only exclude the direct package requested and not its sub-packages but issue seems to say otherwise.
> For example the entire Java SDK packages could be excluded by specifying java. and javax.

If we did do subpackages too, user couldn't use this option on a common package in their project.
@romani Please confirm which way you think.

Either way, we should have a test for this, if we don't already, and some documentation in the xdoc explaining this behavior.